modules/ROOT/pages/tutorials/nbm-selection-1.adoc

```diff
@@ -503,146 +503,160 @@ import org.openide.util.NbBundle;
+files - they work across different NetBeans versions, and alternative editors (like the form editor) still work with all
+components and actions that work with Java files.
 
+This is how NetBeans IDE works with Java and other source files. The thing available from the editor's Lookup is
+a xref:wiki::wiki/DevFaqDataObject.adoc[DataObject], and components like Navigator and Property Sheet watch what object
+the focused `TopComponent` provides.
 
-== Changing Selected Objects on the Fly
-
-To make it really evident how powerful this approach can be, you'll take one more step, and add a button to your editor component that lets it replace the `Event` it has with a new one on the fly.
+This approach is also valuable when migrating existing applications to the NetBeans Platform. The data model object is
+probably existing, working code that shouldn't change for NetBeans integration. By keeping the data model API in a
+separate module, NetBeans integration stays separate from core business logic.
 
 
-[start=1]
-1. Open `MyEditor` in the form editor (click the Design toolbar button in the editor toolbar), and drag a `Button` (javax.swing.JButton) to it.
-
-[start=2]
-1. Set the `text` property of the JButton to "Replace".
+== Changing Selected Objects on the Fly
 
-image::tutorials/selection-1_replace1.png[]
+To show how powerful this approach is, you'll add a button to your editor that replaces the `Event` with a new one
+on the fly.
 
-[start=3]
-1. Right click the `JButton` and choose Events > Action > actionPerformed. 
 
-image::tutorials/selection-1_replace2.png[]
+1. Open `MyEditor` in the form editor (click the Design button) and drag a `Button` (javax.swing.JButton) to it.
 
-This will cause the code editor to open with the cursor in an event handler method. Make that method call `updateContent()`.
+2. Set the button's `text` property to "Replace".
 
+3. Right click the `JButton` and choose Events > Action > actionPerformed. 
++
+This opens the code editor with the cursor in an event handler method. Make the method call `updateContent()`:
++
 [source,java]
 ----
     private void jButton1ActionPerformed(java.awt.event.ActionEvent evt) {                                         
         updateContent();
     }
 ----
-
-Implement the missing method:
-
++
+Add the missing method:
++
 [source,java]
 ----
-    private void updateContent() {
-        Event obj = new Event();
-        jTextField1.setText ("Event #" + obj.getIndex());
-        jTextField2.setText ("Created: " + obj.getDate());
-        setDisplayName ("MyEditor " + obj.getIndex());
-        content.set(Collections.singleton (obj), null);
-    }
+private void updateContent() {
+    Event obj = new Event();
+    jTextField1.setText ("Event #" + obj.getIndex());
+    jTextField2.setText ("Created: " + obj.getDate());
+    setDisplayName ("MyEditor " + obj.getIndex());
+    content.set(Collections.singleton (obj), null);
+}
```

Comment:
we could also remove the space before the `(` to align it with the default formatting settings
```
    Event obj = new Event();
    jTextField1.setText("Event #" + obj.getIndex());
    jTextField2.setText("Created: " + obj.getDate());
    setDisplayName("MyEditor " + obj.getIndex());
    content.set(Collections.singleton(obj), null);
```